src/core/entities/crux-gateway.ts

```diff
@@ -16,21 +16,22 @@ export class CruxGateway {
         if (params.selfChannel && !params.selfChannel.keyManager) {
             throw Error("selfChannel must have keyManager");
         }
+        const messageId = "123e4567-e89b-12d3-a456-426614174000"; // generate
         this.selfChannel = params.selfChannel;
         this.recepientChannel = params.recipientChannel;
         this.messageListener = (message) => undefined;
-        this.packetManager = new GatewayPacketManager(params.protocolHandler, params.selfChannel ? {
+        this.packetManager = new GatewayPacketManager(messageId, params.protocolHandler, params.selfChannel ? {
             cruxId: params.selfChannel.cruxId,
             keyManager: params.selfChannel.keyManager!,
         } : undefined);
     }
 
-    public sendMessage(message: any) {
+    public async sendMessage(message: any) {
```

Comment:
Fix all callers of this function to do `await`

Reply:
> Fix all callers of this function to do `await`

fixed in next PR